contracts/libraries/NftCheck.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+library NftCheck {
+    bytes4 public constant IERC165_ID = type(IERC165).interfaceId;
+    bytes4 public constant IERC1155_ID = type(IERC1155).interfaceId;
+    bytes4 public constant IERC721_ID = type(IERC721).interfaceId;
+
+    function supports721(address collectionAddress) internal view returns(bool) {
+        return _supportsInterface(collectionAddress, IERC721_ID);
+    }
+
+    function supports1155(address collectionAddress) internal view returns(bool) {
+        return _supportsInterface(collectionAddress, IERC1155_ID);
+    }
+
+    function _supportsInterface(address account, bytes4 interfaceId) private view returns (bool) {
+        bytes memory encodedParams = abi.encodeWithSelector(IERC165.supportsInterface.selector, interfaceId);
+        (bool success, bytes memory result) = account.staticcall{gas: 30000}(encodedParams);
```

Comment:
not sure about this being hardcoded at 30000. If people do hardcode this number I am curious to know why that is the case. maybe that is the gas limit and it will always work or something? Not sure

Reply:
yea it's from EIP-165, that's the min gas required i guess https://eips.ethereum.org/EIPS/eip-165